addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -92,12 +92,12 @@ function insert(editor, data, isText = true) {
         fakeEl.replaceChildren(...p.childNodes);
     } else if (fakeEl.childElementCount > 1) {
         // Grab the content of the first child block and isolate it.
-        if (isBlock(fakeEl.firstChild)) {
+        if (isBlock(fakeEl.firstChild) && fakeEl.firstChild.nodeName !== 'TABLE') {
             fakeElFirstChild.replaceChildren(...fakeEl.firstElementChild.childNodes);
             fakeEl.firstElementChild.remove();
         }
         // Grab the content of the last child block and isolate it.
-        if (isBlock(fakeEl.lastChild)) {
+        if (isBlock(fakeEl.lastChild) && fakeEl.lastChild.nodeName !== 'TABLE') {
```

Comment:
I could merge it as is in the meantime to solve the bug at hand, but this looks very ad-hoc and has me wondering if there wouldn't be a deeper issue. In other words: I'd be surprised if we couldn't trigger a similar bug with other elements than table and for which this fix would be insufficient.

Reply:
I just tested this theory and noticed that indeed we have the same issue with lists. So the `isBlock` heuristic is just insufficient. We at least have to add `&& !['TABLE', 'UL', 'OL'].includes(fakeEl.lastChild.nodeName)` but indeed there might be other special cases that we don't want to merge like this, though I can't think of another.

Reply:
@dmo-odoo I pushed the branch with that change.